### PR TITLE
chore: added contributing guide

### LINKED ISSUE #19: Add How to Contribute
Add a Contribution guide MD file to the repository

### PR DESCRIPTION
**Pull request by TapaiBalazs**
Added CONTRIBUTING.md

closes #19

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+### After your pull request is merged
+
+After your pull request is merged, you can safely delete your branch and pull the changes
+from the main (upstream) repository:
+
+- Delete the remote branch on GitHub either through the GitHub web UI or your local shell as follows:
+
+  ```shell
+  git push origin --delete my-fix-branch
+  ```
```
Comment (ktrz) [inline CONTRIBUTING.md:70]: I think we should have auto-deletion of merged branches so this is not necessary

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+- Check out the master branch:
+
+  ```shell
+  git checkout master -f
+  ```
```
Comment (ktrz) [inline CONTRIBUTING.md:76]:
```suggestion
- Check out the main branch:

  ```shell
  git checkout main -f
  ```
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+  ```shell
+  git branch -D my-fix-branch
+  ```
+
+- Update your master with the latest upstream version:
+
+  ```shell
+  git pull --ff upstream master
+  ```
```
Comment (ktrz) [inline CONTRIBUTING.md:88]:
```suggestion
- Update your main with the latest upstream version:

  ```shell
  git pull --ff upstream main
  ```
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+## Submitting a Pull Request (PR)
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+- Search [GitHub](https://github.com/thisdot/open-source/pulls) for an open or closed PR
+  that relates to your submission. You don't want to duplicate effort.
+- Make your changes in a new git branch:
+
+  ```shell
+  git checkout -b my-fix-branch master
```
Comment (ktrz) [inline CONTRIBUTING.md:24]:
```suggestion
  git checkout -b my-fix-branch main
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+- In GitHub, send a pull request to the `this-dot/open-source:main` branch.
+- If we suggest changes then:
+
+  - Make the required updates.
+  - Re-run the test suites to ensure tests are still passing.
+  - Rebase your branch and force push to your GitHub repository (this will update your Pull Request):
+
+    ```shell
+    git rebase master -i
```
Comment (ktrz) [inline CONTRIBUTING.md:53]:
```suggestion
    git rebase main -i
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (TapaiBalazs) [reply CONTRIBUTING.md:70]: should we remove it then?

**Review by ktrz — APPROVED** [review]
LGTM :shipit:

**Pull request merged**